File: backend/internal/models/congress.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

type Congress struct {
	ID        uuid.UUID `json:"id" gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	Title     string    `json:"title" gorm:"not null"`
	Subtitle  string    `json:"subtitle"`
	Description string  `json:"description"`
	Edition   string    `json:"edition"`

	StartDate time.Time `json:"start_date" gorm:"not null"`
	EndDate   time.Time `json:"end_date" gorm:"not null"`

	Location string `json:"location" gorm:"not null"`
	City     string `json:"city"`
	Country  string `json:"country"`

	// Organizational structure as JSON (president, VP, committees, etc.)
	OrganisationalStructure datatypes.JSON `json:"organisational_structure" gorm:"type:jsonb"`

	// Full configuration as JSON (themes, submission types, pricing, deadlines, etc.)
	Config datatypes.JSON `json:"config" gorm:"type:jsonb"`

	// Badge configuration as JSON (template, fields, etc.)
	BadgeConfig datatypes.JSON `json:"badge_config" gorm:"type:jsonb"`

	// The congress admin user ID (auto-created)
	AdminID *uuid.UUID `json:"admin_id" gorm:"type:uuid"`

	// Whether attestations are available for download by participants
	AttestationsAvailable bool `json:"attestations_available" gorm:"default:false"`

	// Status: draft, active, completed, cancelled
	Status string `json:"status" gorm:"default:'draft'"`

	// Super admin who created this congress
	SuperAdminID uuid.UUID `json:"super_admin_id" gorm:"type:uuid;not null"`
	SuperAdmin   User      `json:"-" gorm:"foreignKey:SuperAdminID"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: mobile/lib/models/congress.dart
class Congress {
  final String id;
  final String title;
  final String? subtitle;
  final String? description;
  final String? edition;
  final DateTime startDate;
  final DateTime endDate;
  final String location;
  final String? city;
  final String? country;
  final String status;
  final bool attestationsAvailable;
  final Map<String, dynamic>? config;

  Congress({
    required this.id,
    required this.title,
    this.subtitle,
    this.description,
    this.edition,
    required this.startDate,
    required this.endDate,
    required this.location,
    this.city,
    this.country,
    required this.status,
    required this.attestationsAvailable,
    this.config,
  });

  String get dateRange =>
      '${_fmt(startDate)} - ${_fmt(endDate)}';
  String get fullLocation =>
      [location, city, country].where((e) => e != null && e.isNotEmpty).join(', ');

  static String _fmt(DateTime d) =>
      '${d.day.toString().padLeft(2, '0')}/${d.month.toString().padLeft(2, '0')}/${d.year}';

  bool get isActive => status == 'active';

  factory Congress.fromJson(Map<String, dynamic> json) {
    return Congress(
      id: json['id']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      subtitle: json['subtitle']?.toString(),
      description: json['description']?.toString(),
      edition: json['edition']?.toString(),
      startDate: json['start_date'] != null
          ? DateTime.tryParse(json['start_date'].toString()) ?? DateTime.now()
          : DateTime.now(),
      endDate: json['end_date'] != null
          ? DateTime.tryParse(json['end_date'].toString()) ?? DateTime.now()
          : DateTime.now(),
      location: json['location']?.toString() ?? '',
      city: json['city']?.toString(),
      country: json['country']?.toString(),
      status: json['status']?.toString() ?? 'draft',
      attestationsAvailable: json['attestations_available'] == true,
      config: json['config'] is Map ? json['config'] as Map<String, dynamic>? : null,
    );
  }
}
